desc/protoparse/validate.go

```diff
@@ -0,0 +1,347 @@
+package protoparse
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/golang/protobuf/proto"
+
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+)
+
+func basicValidate(res *parseResult, containsErrors bool) {
```

Comment:
nit: `validateBasic`?

Reply:
Sure... this was already called this FWIW (just moved from `parser.go` into its own file.

Reply:
Ah yea, I was going to add the caveat that this might already have been there - people commenting on something that was just moved around as part of a refactoring PRs is a pet peeve of mine, so feel free to ignore this :-)

Reply:
renamed